server/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -171,6 +174,58 @@ public String toString() {
     public static final String INDICES_FOLDER = "indices";
     public static final String NODE_LOCK_FILENAME = "node.lock";
 
+    public static class NodeLock implements Releasable {
+
+        private final int nodeId;
+        private final Lock[] locks;
+        private final NodePath[] nodePaths;
+
+        public NodeLock(final int nodeId, final Logger logger,
```

Comment:
This looks good to me but I would like another pair of eyes, as it's quite important not to break this. @ywelsch WDYT?